python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py

```diff
@@ -106,33 +109,35 @@ def _should_use_built_artifacts(self) -> bool:
             bool(os.getenv("DAGSTER_DBT_PARSE_PROJECT_ON_LOAD"))
         )
 
-    @property
-    def _current_project_dir(self) -> Path:
-        if self._should_use_built_artifacts():
-            return self._packaged_project_dir if self._packaged_project_dir else self.project_dir
-
-        return self.project_dir
-
-    def _prepare_manifest(self, level: int = logging.DEBUG) -> Path:
+    @public
+    def create_manifest(self, log_level: int = logging.DEBUG) -> Path:
```

Comment:
I'd avoid `compile` only because dbt had the `dbt compile` command, which we suggest avoiding in favor of `dbt parse` in Dagster University. Maybe `generate` could work too?

Reply:
Here's what we want to create on behalf of the user: https://docs.getdbt.com/reference/artifacts/dbt-artifacts

Multiple commands can create these dbt artifacts. They're usually created on dbt invocation (e.g. `dbt run`, `dbt build`, `dbt parse`, `dbt compile`).

To our users, we recommend that they create these artifacts using a command that doesn't execute their models (e.g. `dbt parse`, `dbt compile`). 

The difference between `dbt parse` and `dbt compile` is that `dbt compile` requires your actual credentials from your `profiles.yml` so that it can populate the manifest with metadata from your warehouse, as well as compile your models from jinja to actual SQL. As a result, there's more metadata that you can get in your dbt artifacts.

Some naming suggestions:
- `generate_dbt_artifacts`
- `produce_dbt_artifacts`

Reply:
currently it is `generate_artifacts`, given this is on `DbtProject` do you think the `_dbt_` qualifier is needed?

Reply:
yeah could drop

Reply:
we have a `get_artifact` method on `DbtCliResource`, so that would be inline with that:  https://github.com/dagster-io/dagster/blob/12a03cb890a39099ec6f61f9a789b483e0630a74/python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py#L642-L650